### PR TITLE
STY/TST: update style, flake8-docstrings and hacking in tests, NEP29 (June 21, 2022)

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

- Updates the docstring style following pysat 3.0.2.  
- Implements tests for `flake8-docstrings` and `hacking`
- Bumps NEP 29 versions (June 21, 2022)
- Update of local variable names to snake case in tests

Note that some docstrings may still need to be scrubbed for parameters.

Using June 21, 2022 limits for NEP 29 since RC will not be released before then.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

`flake8 . --count --select=D,E,F,H,W --show-source --statistics`

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: sami2py/_core.py [hunk 1]
```diff
@@ -282,9 +269,27 @@ def run_model(tag='model_run', lat=0, lon=0, alt=300, year=2018, day=1,
...
+    """Generate the information regarding the ExB drifts used by the model.
+
     This information is later stored in the namelist file for SAMI2
+
+    Parameters
+    ----------
+    fejer : bool
+        Specifies whether Fejer-Scherliess model is used
+        If False, then 'exb.inp' is also archived
+    ExB_drifts : 10x2 ndarray of floats, string, or NoneType
```
Comment (JonathonMSmith) [inline sami2py/_core.py:281]:
```suggestion
    exb_drifts : 10x2 ndarray of float, str, or NoneType
```

File: sami2py/_core.py [hunk 2]
```diff
@@ -282,9 +269,27 @@ def run_model(tag='model_run', lat=0, lon=0, alt=300, year=2018, day=1,
...
+        Matrix of Fourier series coefficients dependent on solar local time
+        (SLT) in hours where
+        ExB_total = ExB_drifts[i,0] * cos((i + 1) * pi * SLT / 12)
```
Comment (JonathonMSmith) [inline sami2py/_core.py:284]:
```suggestion
        exb_total = exb_drifts[i,0] * cos((i + 1) * pi * SLT / 12)
```

Comment (JonathonMSmith) [reply sami2py/_core.py:284]: at this point I realize if we want to make this change, then a find and replace is probably better.

File: sami2py/_core_class.py [hunk 3]
```diff
@@ -291,28 +284,30 @@ def _load_model(self):
                                 'long_name': 'neutral wind velocity'})
 
         if self.MetaData['ExB model'] == 'Fourier Series':
```
Comment (JonathonMSmith) [inline sami2py/_core_class.py:286]: This should remain "ExB" right?

Comment (jklenzing) [reply sami2py/_core_class.py:286]: Yes.  The ExB stuff is fixed downstream, you'll see it in the next pull.

**Review by JonathonMSmith — APPROVED** [review]

**Pull request merged**